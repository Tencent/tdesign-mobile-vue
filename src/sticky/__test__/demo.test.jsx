/**
 * 该文件为由脚本 `npm run test:demo` 自动生成，如需修改，执行脚本命令即可。请勿手写直接修改，否则会被覆盖
 */

import { mount } from '@vue/test-utils';
import baseVue from '@/sticky/demos/base.vue';
import containerVue from '@/sticky/demos/container.vue';
import mobileVue from '@/sticky/demos/mobile.vue';
import offsetVue from '@/sticky/demos/offset.vue';

const mapper = {
  baseVue,
  containerVue,
  mobileVue,
  offsetVue,
};

describe('Sticky', () => {
  Object.keys(mapper).forEach((demoName) => {
    it(`Sticky ${demoName} demo works fine`, () => {
      const wrapper = mount(mapper[demoName]);
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
