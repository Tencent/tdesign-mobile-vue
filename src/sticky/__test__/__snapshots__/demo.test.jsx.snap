// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Sticky > Sticky baseVue demo works fine 1`] = `
<div
  class="t-sticky"
  data-v-4a742626=""
  style="height: 0px;"
>
  <div
    class="t-sticky__content"
    style="z-index: 99; position: fixed; top: 0px;"
  >
    
    <button
      aria-disabled="false"
      class="t-button t-button--size-large t-button--base t-button--primary t-button--rectangle common-button"
      data-v-4a742626=""
      role="button"
      type="button"
    >
      <!---->
      <span
        class="t-button__content"
      >
        
        基础吸顶
        
      </span>
      <!---->
    </button>
    
  </div>
</div>
`;

exports[`Sticky > Sticky containerVue demo works fine 1`] = `
<div
  class="sticky-demo"
  data-v-046295d6=""
  id="container"
>
  <div
    class="t-sticky"
    data-v-046295d6=""
    style="height: 0px;"
  >
    <div
      class="t-sticky__content"
      style="z-index: 99; position: fixed; top: 0px;"
    >
      
      <button
        aria-disabled="false"
        class="t-button t-button--size-large t-button--base t-button--default t-button--rectangle common-button green-button"
        data-v-046295d6=""
        role="button"
        type="button"
      >
        <!---->
        <span
          class="t-button__content"
        >
          
          指定容器
          
        </span>
        <!---->
      </button>
      
    </div>
  </div>
</div>
`;

exports[`Sticky > Sticky mobileVue demo works fine 1`] = `
<div
  class="tdesign-mobile-demo sticky-demo-container"
  data-v-3246353e=""
>
  <h1
    class="title"
    data-v-3246353e=""
  >
    Sticky 吸顶
  </h1>
  <p
    class="summary"
    data-v-3246353e=""
  >
    用于常驻页面顶部的信息、操作展示。
  </p>
  <div
    class="tdesign-mobile-demo-block tdesign-mobile-demo-block_notitle"
    data-v-3246353e=""
  >
    <div
      class="tdesign-mobile-demo-block__header"
    >
      <!--v-if-->
      <p
        class="tdesign-mobile-demo-block__summary tdesign-mobile-demo-block_subtitle"
      >
        基础吸顶
      </p>
    </div>
    <div
      class="tdesign-mobile-demo-block__slot"
    >
      
      <div
        class="t-sticky"
        data-v-3246353e=""
        data-v-4a742626=""
        style="height: 0px;"
      >
        <div
          class="t-sticky__content"
          style="z-index: 99; position: fixed; top: 0px;"
        >
          
          <button
            aria-disabled="false"
            class="t-button t-button--size-large t-button--base t-button--primary t-button--rectangle common-button"
            data-v-4a742626=""
            role="button"
            type="button"
          >
            <!---->
            <span
              class="t-button__content"
            >
              
              基础吸顶
              
            </span>
            <!---->
          </button>
          
        </div>
      </div>
      
    </div>
  </div>
  <div
    class="tdesign-mobile-demo-block tdesign-mobile-demo-block_notitle"
    data-v-3246353e=""
  >
    <div
      class="tdesign-mobile-demo-block__header"
    >
      <!--v-if-->
      <p
        class="tdesign-mobile-demo-block__summary tdesign-mobile-demo-block_subtitle"
      >
        吸顶距离
      </p>
    </div>
    <div
      class="tdesign-mobile-demo-block__slot"
    >
      
      <div
        class="t-sticky"
        data-v-3246353e=""
        data-v-ec54aea2=""
        style="height: 0px;"
      >
        <div
          class="t-sticky__content"
          style="z-index: 99; position: fixed; top: 40px;"
        >
          
          <button
            aria-disabled="false"
            class="t-button t-button--size-large t-button--base t-button--danger t-button--rectangle common-button"
            data-v-ec54aea2=""
            role="button"
            type="button"
          >
            <!---->
            <span
              class="t-button__content"
            >
              
              吸顶距离
              
            </span>
            <!---->
          </button>
          
        </div>
      </div>
      
    </div>
  </div>
  <div
    class="tdesign-mobile-demo-block tdesign-mobile-demo-block_notitle"
    data-v-3246353e=""
  >
    <div
      class="tdesign-mobile-demo-block__header"
    >
      <!--v-if-->
      <p
        class="tdesign-mobile-demo-block__summary tdesign-mobile-demo-block_subtitle"
      >
        指定容器
      </p>
    </div>
    <div
      class="tdesign-mobile-demo-block__slot"
    >
      
      <div
        class="sticky-demo"
        data-v-046295d6=""
        data-v-3246353e=""
        id="container"
      >
        <div
          class="t-sticky"
          data-v-046295d6=""
          style="height: 0px;"
        >
          <div
            class="t-sticky__content"
            style="z-index: 99; position: fixed; top: 0px;"
          >
            
            <button
              aria-disabled="false"
              class="t-button t-button--size-large t-button--base t-button--default t-button--rectangle common-button green-button"
              data-v-046295d6=""
              role="button"
              type="button"
            >
              <!---->
              <span
                class="t-button__content"
              >
                
                指定容器
                
              </span>
              <!---->
            </button>
            
          </div>
        </div>
      </div>
      
    </div>
  </div>
</div>
`;

exports[`Sticky > Sticky offsetVue demo works fine 1`] = `
<div
  class="t-sticky"
  data-v-ec54aea2=""
  style="height: 0px;"
>
  <div
    class="t-sticky__content"
    style="z-index: 99; position: fixed; top: 40px;"
  >
    
    <button
      aria-disabled="false"
      class="t-button t-button--size-large t-button--base t-button--danger t-button--rectangle common-button"
      data-v-ec54aea2=""
      role="button"
      type="button"
    >
      <!---->
      <span
        class="t-button__content"
      >
        
        吸顶距离
        
      </span>
      <!---->
    </button>
    
  </div>
</div>
`;
